src/NuGet.Clients/PackageManagement.VisualStudio/VSSettings.cs

```diff
@@ -13,7 +13,23 @@ namespace NuGet.PackageManagement.VisualStudio
     [PartCreationPolicy(CreationPolicy.Shared)]
     public class VSSettings : Configuration.ISettings
     {
-        private Configuration.ISettings SolutionSettings { get; set; }
+        // to initialize SolutionSettings first time outside MEF constructor
+        private Configuration.ISettings _solutionSettings;
+
+        private Configuration.ISettings SolutionSettings
```

Comment:
As far as I can see, `ResetSolutionSettings` doesn't UI thread dependency explicitly, it calls some APIs which requires UI thread but then they all uses `JTF.Run` to switch over to UI thread and executes. So I don't think we should add `ThrowIfNotOnUIThread` since it could be executed on UI thead or BG thread independently.